src/main/python/pybuilder/plugins/python/coverage_plugin.py

```diff
@@ -92,31 +94,41 @@ def do_coverage(project, logger, reactor, execution_prefix, execution_name, targ
     coverage = coverage_factory(cover_pylib=False, branch=True, source=[source_tree_path])
 
     try:
-        project.set_property('__running_coverage',
-                             True)  # tell other plugins that we are not really unit testing right now
-        _start_coverage(coverage)
+        _start_coverage(project, coverage)
 
         if shortest_plan:
             reactor.execute_task_shortest_plan(target_task)
         else:
             reactor.execute_task(target_task)
     finally:
-        _stop_coverage(coverage)
-        project.set_property('__running_coverage', False)
+        _stop_coverage(project, coverage)
 
-    coverage_too_low = False
-    threshold = project.get_property("%s_threshold_warn" % execution_prefix)
-    exceptions = project.get_property("%s_exceptions" % execution_prefix)
+    module_exceptions = project.get_property("%s_exceptions" % execution_prefix)
+    modules = _list_all_covered_modules(logger, module_names, module_exceptions)
 
-    report = {
-        "module_names": []
-    }
+    failure = _build_coverage_report(project, logger, execution_name, execution_prefix, coverage, modules)
+    if failure:
```

Comment:
Thanks, but that's simply to make sure the report builder wasn't controlling the execution flow and delegated the failure decision to a parent.